Rules/AvoidPositionalParameters.cs

```diff
@@ -27,6 +27,18 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
         {
             if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
 
+            // Find all function definitions in the script and add them to the set.
+            IEnumerable<Ast> functionDefinitionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);
+            HashSet<String> declaredFunctionNames = new HashSet<String>();
+
+            foreach (Ast foundAst in functionDefinitionAsts) {
+                FunctionDefinitionAst functionDefinitionAst = (FunctionDefinitionAst)foundAst;

```

Comment:
I think you can inline the cast in the foreach loop: `foreach (FunctionDefinitionAst functionDefinitionAst in functionDefinitionAsts)`